keps/sig-cli/kustomize-default-config.md

```diff
@@ -0,0 +1,314 @@
+---
+title: Kustomize Default Configuration Adjustments
+authors:
+ - "@jbrette"
+owning-sig: sig-cli
+participating-sigs:
+ - sig-apps
+ - sig-api-machinery
+reviewers:
+ - "@monopole"
+ - "@Liujingfang1"
+approvers:
+ - "@Liujingfang1"
+editors:
+ - "@jbrette"
+creation-date: 2019-09-01
+last-updated: 2019-09-01
+status: provisional
+see-also:
+replaces:
+superseded-by:
+ - n/a
+---
+
+# Kustomize Default Configuration Adjustments
+
+## Table of Contents
+* [Table of Contents](#table-of-contents)
+* [Summary](#summary)
+* [Motivation](#motivation)
+  * [Goals](#goals)
+  * [Non-Goals](#non-goals)
+* [Proposal](#proposal)
+  * [Implementation Details/Notes/Constraints](#implementation-detailsnotesconstraints)
+  * [Risks and Mitigations](#risks-and-mitigations)
+* [Graduation Criteria](#graduation-criteria)
+* [Implementation History](#implementation-history)
+* [Alternatives](#alternatives)
+
+[Tools for generating]: https://github.com/ekalinin/github-markdown-toc
+
+## Summary
+
+Kustomize builtin transformers are provided with default configurations describing which
+resource fields are subject of a particular transformation. kustomize provides the ability
+to add fields to the default configuration but does not let the user adjust that configuration
+in order to skip a transformation for a specific resource or resource kind.
+
+We are aiming at addressing those issues:
+- Make cluster level kind configurable [#617](https://github.com/kubernetes-sigs/kustomize/issues/617)
+- Do not add namespace to cluster scoped CRDs [#552](https://github.com/kubernetes-sigs/kustomize/issues/552) 
+- Skipping nameprefix for some resource kind [#519](https://github.com/kubernetes-sigs/kustomize/issues/617)
+- Adding labels only to metadata.labels [#330](https://github.com/kubernetes-sigs/kustomize/issues/330)
+- Skip adding labels to certain paths [#519](https://github.com/kubernetes-sigs/kustomize/issues/519)
+- Unable to disable commonLabels injection using transformer config: Error: conflicting fieldspecs [#817](https://github.com/kubernetes-sigs/kustomize/issues/817)
+
+## Motivation
+
+
+### Goals
+
+- Provide the ability for the user to skip transformation for specific group/version/kind especially usefull if the default
+  configuration is using a wild card (see namespace and prefixsuffix transformer configuration).
+- Provide the ability of the user to replace or remove an entry in the default configuration which is not matching his needs
+  (see commonLabel transformer configuration)
+
+  
+### Non-Goals
+
+- Provide backward compatibility with the transformer default configurations.
```

Comment:
We should keep backward compatibility otherwise who customize the builtin transformer configuration will experience breaking changes.

Keep backward compatibility should be listed as a Goal